src/RecordProfileForm.js

```diff
@@ -0,0 +1,78 @@
+import React, { useReducer } from 'react';
+import Form from 'react-bootstrap/Form';
+import Button from 'react-bootstrap/Button';
+import FormSelect from './FormSelect';
+import { db } from './Firebase';
+
+export default () => {
+  const [state, update] = useReducer(
+    (state, update) => ({ ...state, ...update }),
+    {
+      firstName: '',
+      lastName: '',
+    }
+  );
+
+  const handleChange = e => {
+    const value = e.target.value;
+    const id = e.target.id;
+    update({ [id]: value });
+  };
+
+  const handleSubmit = async event => {
+    event.preventDefault();
+    event.stopPropagation();
+
+    try {
+      await db.collection('participants').add({
```

Comment:
Here you want to update the user you read from earlier rather than add to the table:

`await userDoc.update({firstName, lastName})`

you can test that this works by refreshing the page. The name should reflect the updated value.

Reply:
Haven't tested this yet due to issues with comment above.

Reply:
works now :)